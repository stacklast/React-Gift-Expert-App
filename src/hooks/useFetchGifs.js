import { useEffect, useState } from "react"
import { getGifts } from '../helpers/getGifs'
export const useFetchGifs = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {  

        getGifts( category )
            .then( imgs => {

                setTimeout( ()=> {
                    
                    //console.log(imgs);

                    setState({
                        data:imgs,
                        loading: false
                    });
                }, 3000);

            });

    }, [ category ]);

    

    return state;
}
