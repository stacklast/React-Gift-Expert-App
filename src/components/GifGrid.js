import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem }  from './GifGridItem'

export const GifGrid = ( { category } ) => {

    const { data:images, loading } = useFetchGifs();
    return (
        <>
            <h3 className="animate__animated animate__fadeIn"> { category } </h3>
            { loading && <p>loading</p> }
            <div className="card-grid">

                    {
                        images.map( img => (
                            <GifGridItem
                            key={ img.id } 
                            { ...img }
                            />
                        ))
                    }
            
            </div>
        </>
    )
}
