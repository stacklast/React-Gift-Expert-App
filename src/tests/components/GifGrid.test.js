import * as React from 'react';
import { shallow} from 'enzyme';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba en <GifGrid />', () => {
    const category = 'Batman';
    

    test('debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper =  shallow(<GifGrid category={ category } />);
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('debe mostrar imagenes cuando se carga useFetchGifs', () => {
        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/image.png',
                title: 'cual quier cosa'
            },
            {
                id: '123',
                url: 'https://localhost/image.png',
                title: 'cual quier cosa'
            }
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper =  shallow(<GifGrid category={ category } />);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
    

});
