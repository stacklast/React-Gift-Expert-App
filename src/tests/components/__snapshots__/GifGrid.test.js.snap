// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGrid /> debe mostrar imagenes cuando se carga useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Batman
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cual quier cosa"
      url="https://localhost/image.png"
    />
    <GifGridItem
      id="123"
      key="123"
      title="cual quier cosa"
      url="https://localhost/image.png"
    />
  </div>
</Fragment>
`;

exports[`Prueba en <GifGrid /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Batman
     
  </h3>
  <p>
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
